lib/ex_aws/config/auth_cache.ex

```diff
@@ -72,14 +70,35 @@ defmodule ExAws.Config.AuthCache do
           auth
 
         adapter ->
-          adapter.adapt_auth_config(auth, profile, expiration)
+          attempt_credentials_refresh(adapter, auth, profile, expiration)
       end
 
+    Process.send_after(self(), {:refresh_awscli_config, profile, expiration}, expiration)
     :ets.insert(ets, {{:awscli, profile}, auth})
 
     auth
   end
 
+  defp attempt_credentials_refresh(adapter, auth, profile, expiration, retries \\ 6) do
+    case adapter.adapt_auth_config(auth, profile, expiration) do
+      {:error, error} ->
+        if retries == 1 do
```

Comment:
Rather than a separate `if` block, you could check this condition in the `case` by doing
```
{:error, error} when retries == 1 ->
    ...do stuff
{:error, _error} ->
    do sleep and retry